src/StreamJsonRpc/JsonMessageFormatter.cs

```diff
@@ -403,6 +447,8 @@ private JsonRpcResult ReadResult(JToken json)
             JToken id = json["id"];
             JToken result = json["result"];
 
+            this.ClearProgressObject(id.Value<long>());
```

Comment:
>id.Value<long>() [](start = 37, length = 16)

The result message's ID may not always be an integer. [Some servers](https://github.com/microsoft/vs-streamjsonrpc/pull/291/files#diff-2eba41192bde2ae51a16aa411035dfb4) return our integer as a string. Will `Value<long>()` perform the string-to-long conversion for us? If not, we should add tests for this and fix it. Here and everywhere else you access the ID in this way.
I suggest you just pass `id` in and let `ClearProgressObject` do the cast/conversion in one place.

Reply:
I see you trimmed this down to `id` on other calls to `ClearProgressObject`, but not this one.

---
In reply to: [302242972](https://github.com/milopezc/vs-streamjsonrpc/pull/1#discussion_r302242972) [](ancestors = 302242972)